superset/connectors/sqla/models.py

```diff
@@ -304,7 +306,11 @@ class SqlaTable(Model, BaseDatasource):
     owner_class = security_manager.user_model
 
     __tablename__ = "tables"
-    __table_args__ = (UniqueConstraint("database_id", "table_name"),)
+
+    # Note this unqiuness constraint is not part of the physicalschema, i.e., it doesn't
+    # exist in the migrations, but is required by `import_from_dict` to ensure the
+    # correct filters are applied in order to identify uniqueness.
+    __table_args__ = (UniqueConstraint("database_id", "schema", "table_name"),)
```

Comment:
@mistercrunch I'm really not a fan of this but the [import_from_dict](https://github.com/apache/incubator-superset/blob/648f9fa54b90ce4cca84737a21214fd92c0e9fcc/superset/models/helpers.py#L110) method uses the uniqueness constraint add add additionally filters to match the relevant record in the database.